src/SaveDialog.vala

```diff
@@ -249,4 +260,11 @@ public class Screenshot.SaveDialog : Granite.Dialog {
             folder_image.gicon = new ThemedIcon ("folder");
         }
     }
+
+    public void hide_destroy () {
```

Comment:
Since you're watching for the clipboard change do you still need this function?

Reply:
I'm also not a big fan of timeouts, but at least in my test cases, I haven't been able to make it work without this wait time :(.

Reply:
Can we make it one `on_clipboard_changed` function and add a comment then? Instead of having this split between a lambda and a function